packages/table/__tests__/table-column.spec.ts

```diff
@@ -944,4 +944,139 @@ describe('table column', () => {
+              address: '1518 Jinshajiang Road, Putuo District, Shanghai',
+              index: 5,
+            },
+            {
+              id: 312,
+              date: '2016-05-01',
+              name: 'Wangxiaohu',
+              address: '1518 Jinshajiang Road, Putuo District, Shanghai',
+              index: 6,
+            },
+            {
+              id: 313,
+              date: '2016-05-01',
+              name: 'Wangxiaohu',
+              address: '1518 Jinshajiang Road, Putuo District, Shanghai',
+              index: 7,
+              disabled: true,
+            },
+          ],
+        }, {
+          id: 32,
+          date: '2016-05-01',
+          name: 'Wangxiaohu',
+          address: '1518 Jinshajiang Road, Putuo District, Shanghai',
+          index: 8,
+        }],
+      }, {
+        id: 4,
+        date: '2016-05-03',
+        name: 'Wangxiaohu',
+        address: '1518 Jinshajiang Road, Putuo District, Shanghai',
+        index: 9,
+      }]
+    }
+
+    const createTable = function(methods) {
+      return mount(
+        Object.assign(
+          {
+            components: {
+              ElTable,
+              ElTableColumn,
+            },
+            template: `
+              <el-table
+                ref="table"
+                :data="testData"
+                row-key="id"
+                border
+                default-expand-all
+                :tree-props="{children: 'children', hasChildren: 'hasChildren'}"
+              >
+                <el-table-column type="index"></el-table-column>
+                <el-table-column type="selection" :selectable="selectable"></el-table-column>
+                <el-table-column prop="id" label="id"></el-table-column>
+                <el-table-column
+                  prop="date"
+                  label="Date"
+                  sortable
+                  width="180">
+                </el-table-column>
+                <el-table-column
+                  prop="name"
+                  label="Name"
+                  sortable
+                  width="180">
+                </el-table-column>
+                <el-table-column
+                  prop="address"
+                  label="Address">
+                </el-table-column>
+              </el-table>
+          `,
+            methods: {
+              selectable(row) {
+                return !row.disabled
+              },
+              ...methods,
+            },
+            data() {
+              return {
+                testData: getTableData(),
+              }
+            },
+          },
+        ),
+      )
+    }
+
+    it('selectable index parameter should be correct', async() => {
+      const wrapper = createTable({
+        selectable(row, index) {
+          expect(row.index - 1).toEqual(index)
+          return !row.disabled
+        },
+      })
+      await nextTick()
+      wrapper.vm.$refs.table.toggleAllSelection()
+      wrapper.unmount()
+    })
```

Comment:
use assertion like this:

```ts
const expectToBeCalled = jest.fn()
const wrapper = createTable({
        selectable(row, index) {
          expectToBeCalled((row.index - 1) === index)
          return !row.disabled
        },
      })
await nextTick()
wrapper.vm.$refs.table.toggleAllSelection()

expect(expectToBeCalled).toHaveBeenCalledWith(true)

```

Reply:
I need to make sure that the result is correct every time, but I refactored the code anyway

Reply:
Humm, I think you can do that by asserting your result to `expect(toBeCalled).toHaveBeenNthCalled(order, params)`. Use a for loop to assert your result